### PR TITLE
Added notes on 'developmental mode'

### PR DESCRIPTION
**Pull request by redeboer**

### CODE REVIEW

File: doc/source/contribute.rst [hunk 1]
```diff
@@ -1,130 +1,145 @@
...
+First, go into your pycompwa folder, then un-install the pycompwa install from PyPI (if necessary), and set up the virtual environment wrapper: |br|
+
+.. code-block:: shell
+
+  pip uninstall pycompwa # if necessary
+  pip install virtualenvwrapper
+  source venv/bin/virtualenvwrapper.sh
+  add2virtualenv .
+
+In this example, we assume you use Python's ``venv`` and that you set your virtual environment is set up in a folder called ``venv`` (see 3rd line). You can also work with other virtual environment handlers, such as `Conda <https://docs.conda.io/en/latest/>`__.
```
Comment (spflueger) [inline doc/source/contribute.rst]: Maybe you can just use something like `$VENV: path to your virtual env`. Then it is does not matter what you use and you can skip this info. Actually conda is more than a virtual env handler.

File: doc/source/contribute.rst [hunk 2]
```diff
@@ -1,130 +1,145 @@
...
+
+Now, go through the standard build procedure:
```
Comment (spflueger) [inline doc/source/contribute.rst]: I would scratch standard

File: doc/source/contribute.rst [hunk 3]
```diff
@@ -1,130 +1,145 @@
+.. |br| raw:: html
```
[the remark this answers is not in the document]
Comment (redeboer) [reply doc/source/contribute.rst]: No, the font size becomes smaller in line block syntax. Maybe this behaviour comes from some sort of CSS file?

File: doc/source/contribute.rst [hunk 4]
```diff
@@ -1,130 +1,145 @@
...
-added to the main repository.
+   Note that the remote from which you cloned the repository is named ``origin`` by default (here: your fork). A local ``master`` branch is automatically checked out from the origin after the clone as well. You can list all branches with ``git branch -a``.
+
+You repeat the following steps until your contribution is finished. Only then can your contributions be added main repository through a `pull request <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests>`__.
 
 .. tip::
-   Remember to commit frequently to not have single commits with too many
-   changes! Rebasing can help you later on to group and alter commit messages,
-   so don't worry.
+   Remember to commit frequently instead of submitting a pull request of just one commit. Making frequent snapshots (commits) of your work is safer workflow in general. Later on, rebasing can help you to group and alter commit messages, so don't worry.
```
[the remark this answers is not in the document]
Comment (redeboer) [reply doc/source/contribute.rst]: Yeah we may want to be more specific about our preferences for how to work with commits. The remark here is just a general remark about Git, though.

File: doc/source/contribute.rst [hunk 5]
```diff
@@ -1,130 +1,145 @@
...
-* Enter a meaningful commit message. First line is a overall summary.
-  Then a more detailed description if neccessary.
-* Synchronize with the changes from the main repository/upstream
-  
-  * Fetch new changes ``git fetch upstream``
-  * Reapply your current branch commits to the head of upstream master branch
+* Enter a meaningful commit message. First line is a overall summary. Then, if necessary, skip one line and add a more detailed description form the third line on.
+* Synchronize with the changes from the main repository/upstream:
+
+  - Fetch new changes: |br|
```
[the remark this answers is not in the document]
Comment (redeboer) [reply doc/source/contribute.rst]: Nope

File: doc/source/contribute.rst [hunk 6]
```diff
@@ -1,130 +1,145 @@
...
-  pull request
+* Log into GitHub with your account and create a pull request (merge your fork branch into the main repository master branch)
+* While open, commits pushed to your fork branch will automatically update the pull request
+
+.. _DevelopmentalSetup:
+
+Developmental set-up
+--------------------
+
+If you want to develop new functionality for pycompwa, the :ref:`PyPI setup <InstallFromPyPI>` won't suffice and you will have to :ref:`build from source <InstallFromSource>`. In addition, it is useful to use a `virtual environment wrapper <https://virtualenvwrapper.readthedocs.io/en/latest/>`__ instead of using a local ``pip`` install of pycompwa. This way, you directly call your local pycompwa source code, so you can modify and test things immediately. It is advised to work within a virtual environment for this set-up as well, just as in the normal :ref:`Installation` procedure.
```
[the remark this answers is not in the document]
Comment (redeboer) [reply doc/source/contribute.rst]: True, I will remove the reference

File: doc/source/contribute.rst (hunk 1 above)
Comment (redeboer) [reply doc/source/contribute.rst]: Currently, we only vaguely refer to `pipenv` and `venv`, not yet to Conda. I think we should stick with just one virtual environment handler.

> In either case, it is highly recommended to setup and use a python virtual environment for pycompwa. Use the tool of your liking, popular choices are pipenv and venv. [[link](https://pycompwa.readthedocs.io/en/latest/installation.html)]

File: doc/source/contribute.rst (hunk 2 above)
Comment (redeboer) [reply doc/source/contribute.rst]: > I made some suggestions. The part about referring to building from source in the development mode is the most important one.

I think in general we should simplify the [Installation](https://pycompwa.readthedocs.io/en/latest/installation.html) page: stick with one virtual environmant (Conda?), then specifically write how to set it up, and write (at most) in a note that you can also work with `venv` etc.

The installation page should also be more simple: only explain for instance `pip install` and then refer to the Contribute page for the 'developmental setup' (i.e. build from source).

**Pull request merged**